kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -99,12 +105,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
           LOG.info("Writing to existing dataset/view: " + view);
           break;
         case CHECKPOINT:
-          boolean ready = true; // dataset.isReady(); // TODO: add when CDK-451 is ready
           long lastModForTarget = -1;
           if (view instanceof LastModifiedAccessor) {
             lastModForTarget = ((LastModifiedAccessor) view).getLastModified();
           }
-          if (ready && lastModForTarget > lastModForSource) {
+
+          if (ready && (lastModForTarget > lastModForSource)) {
```

Comment:
That clarifies the comment.

I think what we're differing on is the semantics meaning of the signal. A modification time can be influenced by the ready signal time, but the modification time does not imply readiness. For example if the output is written non-atomically (say in a two part merge) and fails part way through the job fails and the ready signal is not written. A check on the modification time would be influenced by the first completed portion of the non-atomic write, but the job would be incorrect to use that partial output as a complete checkpoint. Treating the ready signal independently protects against that.